app/components/post_list/post/header/expiry_timer/index.tsx

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useEffect, useState} from 'react';
+import {Text, View} from 'react-native';
+
+import CompassIcon from '@components/compass_icon';
+import {useTheme} from '@context/theme';
+import {formatTime} from '@utils/datetime';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+
+import {TIMER_REFRESH_INTERVAL_MS} from './constants';
+
+type Props = {
+    expiryTime: number; // timestamp in ms
+    onExpiry?: () => void;
+};
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => ({
+    container: {
+        display: 'flex',
+        flexDirection: 'row',
+        gap: 6,
+        backgroundColor: changeOpacity(theme.dndIndicator, 0.12),
+        paddingHorizontal: 5,
+        paddingVertical: 2,
+        borderRadius: 4,
+    },
+    text: {
+        fontSize: 12,
+        color: theme.dndIndicator,
+        fontWeight: 600,
+    },
+}));
+
+const ExpiryCountdown: React.FC<Props> = ({expiryTime, onExpiry}) => {
+    const theme = useTheme();
+    const style = getStyleSheet(theme);
+
+    const [remainingSeconds, setRemainingSeconds] = useState(() => Math.max(0, expiryTime - Date.now()) / 1000);
+
+    useEffect(() => {
+        const timer = setInterval(() => {
+            const remainingTime = Math.max(0, expiryTime - Date.now()) / 1000;
```

Comment:
The body of the interval doesn't seem to be covered by the tests.

Reply:
yeah this is taking too long for me for some reason. I will defer it to after this PR.